docs/designs/Compilation.md

```diff
@@ -0,0 +1,20 @@
+# Page compilation
+
+The compilation system for ASP.NET Framework was MSBuild based and not something that we needed to bring-forward as-is as it was mostly an implementation detail. Instead, we are using Roslyn and creating compilations based on that.
+
+## IWebFormsCompiler
+
+The `IWebFormsCompiler` interface is used to access the heavy lifting of taking a collection of input files and compiling that aspx/ascx/etc into their appropriate types. It takes an `ICompilationStrategy` that is used to configure what the compilation should look like. The main implementations of this are for in-memory (i.e. runtime) compilation compared to static compilation (i.e. aspnet_compiler).
```

Comment:
You probably wants to have another section explainaing  ICompilationStrategy. 

Also see if you want to follow [this style](https://github.com/CoreWCF/CoreWCF/blob/30a6ce9c86332bb11f20eccd2ba84404f04b7165/Documentation/DesignDocs/transport_message_cred_using_cert.md#iservicedispatcher).(This is lot of work, so optional :))

Reply:
great style, but feels like too much work right now. The goal of these docs at the moment is to be able to point people to where they should look for more details.